Makefile

```diff
@@ -341,19 +341,21 @@ build-manifest: kustomize
 .PHONY: build-manifest
 
 ##@ Development Env
-CLUSTER_PROVIDER ?= kind
-LOCAL_DEV_CLUSTER_VERSION ?= main
-
-KIND_WORKER_NODES ?=2
-BUILD_CONTAINERIZED ?= build_image
+export CLUSTER_PROVIDER ?= kind
```

Comment:
I think for local dev we should stick container runtime to docker as with podman it won't work as expected. For instance while doing `make cluster_up` as of now it will fail due to the unavailability of `kind` network since podman doesn't create a network automatically if not present, unlike [docker](https://github.com/sustainable-computing-io/local-dev-cluster/blob/a3aaeb684fc94f6c4d167b18d27c5aa8df8f810d/providers/kind/kind.sh#L103).

Reply:
I tested this on my mac with podman desktop and it worked 🤔 maybe something has changed there as I saw the failure you described in CI.

Reply:
That's strange 🤔 Is the `kind` network already present if you do `podman network ls`?

Reply:
```console
$ podman network ls
NETWORK ID    NAME                     DRIVER
24edfc3079df  kind                     bridge
```

I didn't create that network

Reply:
What we (Vibhu & I) observed is that the kind create fails the first time. It however creates `kind` network and fails. Re running `kind create` succeeds the second time

Reply:
@dave-tucker I tried on a fresh fedora using the latest podman and kind. `make cluster-up` failed due to the same issue `network kind was not found` 
```
 ❯ /usr/bin/podman run -d --restart=always --name kind-registry -p 127.0.0.1:5001:5000 --network kind registry:2

Resolved "registry" as an alias (/etc/containers/registries.conf.d/000-shortnames.conf)
Trying to pull docker.io/library/registry:2...
Getting image source signatures
Copying blob bc8f2b8a18ff done   |
Copying blob d49090716641 done   |
Copying blob ad02dd2076d6 done   |
Copying blob 9d41963883ad done   |
Copying blob 73baa7ef167e done   |
Copying config 6a3edb1d5e done   |
Writing manifest to image destination
Error: unable to find network with name or ID kind: network not found
make: *** [Makefile:411: cluster-up] Error 125

❯ podman version
Client:       Podman Engine
Version:      5.1.1
API Version:  5.1.1
Go Version:   go1.22.3
Built:        Tue Jun  4 05:30:00 2024
OS/Arch:      linux/amd64

❯ podman network ls
NETWORK ID    NAME        DRIVER
2f259bab93aa  podman      bridge

❯ kind version
kind v0.23.0 go1.21.10 linux/amd64

❯ podman network ls
NETWORK ID    NAME        DRIVER
2f259bab93aa  podman      bridge
```